build.gradle

```diff
@@ -0,0 +1,245 @@
+    malhar_library: "org.apache.apex:malhar-library:$apex_malhar_version",
+    mockito_core: "org.mockito:mockito-core:1.9.5",
+    netty_handler: "io.netty:netty-handler:$netty_version",
+    netty_tcnative_boringssl_static: "io.netty:netty-tcnative-boringssl-static:1.1.33.Fork26",
+    netty_transport_native_epoll: "io.netty:netty-transport-native-epoll:$netty_version",
+    protobuf_java: "com.google.protobuf:protobuf-java:$protobuf_version",
+    protobuf_java_util: "com.google.protobuf:protobuf-java-util:$protobuf_version",
+    proto_google_cloud_pubsub_v1: "com.google.api.grpc:proto-google-cloud-pubsub-v1:$pubsub_grpc_version",
+    proto_google_cloud_spanner_admin_database_v1: "com.google.api.grpc:proto-google-cloud-spanner-admin-database-v1:$grpc_google_common_protos",
+    proto_google_common_protos: "com.google.api.grpc:proto-google-common-protos:$grpc_google_common_protos",
+    slf4j_api: "org.slf4j:slf4j-api:1.7.25",
+    slf4j_jdk14: "org.slf4j:slf4j-jdk14:1.7.25",
+    slf4j_log4j12: "org.slf4j:slf4j-log4j12:1.7.25",
+    snappy_java: "org.xerial.snappy:snappy-java:1.1.4",
+    spark_core: "org.apache.spark:spark-core_2.10:$spark_version",
+    spark_network_common: "org.apache.spark:spark-network-common_2.10:$spark_version",
+    spark_streaming: "org.apache.spark:spark-streaming_2.10:$spark_version",
+    stax2_api: "org.codehaus.woodstox:stax2-api:3.1.4",
+    woodstox_core_asl: "org.codehaus.woodstox:woodstox-core-asl:4.4.1",
+  ],
+]
+
+buildscript {
+  repositories {
+    mavenLocal()
+    mavenCentral()
+    jcenter()
+    maven { url "https://plugins.gradle.org/m2/" }
+    maven { url "http://repo.spring.io/plugins-release" }
+  }
+  dependencies {
+    classpath "net.ltgt.gradle:gradle-apt-plugin:0.12"
+    classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.1"
+    classpath "io.spring.gradle:propdeps-plugin:0.0.9.RELEASE"
+    classpath "gradle.plugin.org.nosphere.apache:creadur-rat-gradle:0.3.1"
+    classpath "com.commercehub.gradle.plugin:gradle-avro-plugin:0.11.0"
+    classpath "com.diffplug.spotless:spotless-plugin-gradle:3.6.0"
+    classpath "gradle.plugin.com.github.blindpirate:gogradle:0.7.0"
+    classpath "gradle.plugin.com.palantir.gradle.docker:gradle-docker:0.13.0"
+    classpath "cz.malohlava:visteg:1.0.3"
+    classpath "com.github.jengelman.gradle.plugins:shadow:2.0.1"
+  }
+}
+
+/*************************************************************************************************/
+// Configure the root project
+
+apply plugin: "base"
+
+// Apply one top level rat plugin to perform any required analysis
+apply plugin: "org.nosphere.apache.rat"
+rat {
+  plainOutput = true
+  xmlOutput = false
+  htmlOutput = false
+  failOnError = true
+  excludes = [
+    // Exclude files generated by the Gradle build process
+    "**/.gradle/**",
+    "**/.gogradle/**",
+    "**/build/**",
+    "**/vendor/**",
+
+    // .gitignore: Ignore files generated by the Maven build process
```

Comment:
<!--new_thread; commit:5886e58537a4f58de6611a40e5e8e09655913ae0; resolved:0-->
Incidentally, the plugin might already check .gitignore. It was my mistake thinking that the Maven plugin didn't.